File: app/src/main/java/com/example/jaguar/weather/model/WeatherContract.java
package com.example.jaguar.weather.model;

import java.util.List;

public interface WeatherContract {

    interface WeatherModel {
        void updateWeather(String str, CallBack call);
    }
    interface WeatherView {
        void updateAdapter(List<WeatherObj> objs);
        void showProgress();
        void hideProgress();
    }
    interface WeatherPresenter {
        void onSelect(String str);
    }

}

File: app/src/main/java/com/example/jaguar/weather/model/WeatherContract.kt
package com.example.jaguar.weather.model

interface WeatherContract {
    interface WeatherModel {
        fun updateWeather(str: String, call: CallBack)
    }

    interface WeatherView {
        fun updateAdapter(objs: MutableList<WeatherObj>)
        fun showProgress()
        fun hideProgress()
    }

    interface WeatherPresenter {
        fun onSelect(str: String)
    }
}